### PR TITLE
add the --dump-settings command line argument

### PR DESCRIPTION
**Pull request by leotrs**
Addresses [#1762](https://github.com/getpelican/pelican/issues/1762).

### CODE REVIEW

File: pelican/__init__.py [hunk 1]
```diff
@@ -327,6 +327,10 @@ def parse_arguments():
                         help=('Exit the program with non-zero status if any '
                               'errors/warnings encountered.'))
 
+    parser.add_argument('--dump-settings', dest='dump_settings',
+                        action='append', nargs='?', help='Read'
+                        'config file, dump final settings and exit.')
```
Comment (Scheirle) [inline pelican/__init__.py]: 1. There is no space between `Read` and `config`.
2. The help text does not describe what the optional parameter does.

File: pelican/__init__.py [hunk 2]
```diff
@@ -390,6 +394,19 @@ def main():
...
+        if args.dump_settings:
+            from pprint import pprint
+
+            # 'pelican --dump-settings': dump_settings = [None]
+            if args.dump_settings[0] is None:
+                pprint(settings)
+
+            # 'pelican --dump-settings SETTING': dump_settings = [SETTING]
+            else:
+                pprint(settings[args.dump_settings[0]])
```
Comment (Scheirle) [inline pelican/__init__.py]: A nice warning/error instead of a `KeyError` exception would be nice if the given argument is not in the settings dict.

**Comment by justinmayer** [comment]
@leotrs: Any chance you could make some modifications based on the feedback here?

**Comment by justinmayer** [comment]
Hi Leo. Just checking in regarding your contribution to Pelican. Could you have a look at the feedback @Scheirle provided and update your pull request with any changes you feel are warranted?

**Comment by justinmayer** [comment]
Hey Leo. Would like to get this merged. Would you please have a look at the feedback @Scheirle provided and update your pull request with any changes you feel are warranted?

**Comment by justinmayer** [comment]
Hi Leo. Any update on your contribution here?

**Comment by justinmayer** [comment]
Superseded by above-referenced PR.

**Pull request closed without merging**